sm/download/src/download.rs

```diff
@@ -71,16 +79,43 @@ impl Downloader {
             }
         })
         .await?;
+        let mut response = response;
+        let mut file = File::create(self.target_filename.as_path())?;
 
-        let content = response.bytes().await?;
+        if let Some(file_len) = response.content_length() {
+            let tmpstats = statvfs::statvfs("/tmp")?;
+            // Reserve 5% of total disk space
+            let five_percent_disk_space = (tmpstats.blocks() * tmpstats.block_size()) * 5 / 100;
+            let usable_disk_space =
+                tmpstats.blocks_free() * tmpstats.block_size() - five_percent_disk_space;
 
-        // Cleanup after `disc full` will happen inside atomic write function.
-        tedge_utils::fs::atomically_write_file_async(
-            self.download_target.as_path(),
-            self.target_filename.as_path(),
-            content.as_ref(),
-        )
-        .await?;
+            if file_len >= usable_disk_space {
+                return Err(DownloadError::InsufficientSpace);
+            }
+
+            let free_space_after_download = usable_disk_space - file_len;
+            if five_percent_disk_space > free_space_after_download {
+                return Err(DownloadError::InsufficientSpace);
+            }
```

Comment:
This check must be removed.

* As currently implemented, this check doubles the amount of space kept free (since `five_percent_disk_space` is already accounted by `usable_disk_space`).
* If fixed, the check is useless (since checking that a download can be done only if  `len` < `free` - `five_percent_disk_space `).

Reply:
Agreed, fixed it.